\documentclass{bioinfo}
\copyrightyear{2015}
\pubyear{2015}

\begin{document}
\firstpage{1}

\title[short Title]{Pyveplot: SVG Hiveplot API}
\author[Sample \textit{et~al}]{Rodrigo Garc'ia-Herrera\,$^{1,*}$ \footnote{to whom correspondence should be addressed}}
\address{$^{1}$Department of Bioinformatics, Mexican Institute of
  Genomic Medicine}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\maketitle

\begin{abstract}

\section{Sumary:}
Python library provides programmatic object oriented interface for the creation of
HivePlots in Scalable Vector Graphics format.
\section{Availability and Implementation:}
Freely available as a Python package at
https://pypi.python.org/pypi/pyveplot/

\section{Contact:} \href{rgarcia@inmegen.gob.mx}{rgarcia@inmegen.gob.mx}
\end{abstract}

\section{Introduction}

Hiveplots are a way of visualizing large networks which allow
for easy assesment and comparison of their structural properties. \citealp{Boffelli03}

Scalable Vector Graphics (SVG) is a modularized language for
describing two-dimensional vector graphics, and an open standard
recomended by the World Wide Web Consortium (W3C).
\citealp{McCormack:11:SVG} 


Although the requirements of a Hiveplot can be met
with the most basic shapes supported by SVG, its capabilities can
be exploited much further.




Though tools are available for the creation of Hiveplots they tend to
be graphical user interfaces which may not be flexible enough, or
programmatic interfaces which may not be compatible

Python is a mature and popular programming language with a large
number of libraries related to data analysis in general and life
sciences and bioinformatics in particular.

This library provides a procedural, imperative approach to the
assembly of a plot, which may have better readability than a more
declarative approach.

\section{Object Oriented Approach}

The Object Oriented (OO) approach to the Aplication Programming
Interface (API) makes the creation of a plot a straightforward
process. A hiveplot consists of 
\begin{itemize}
\item radialy distributed linear axes
\item nodes along those axes
\item conections among those nodes
\end{itemize}
The API provides the corresponding objects: a Hiveplot object which
contains an arbitrary number of Axis objects which in turn contain an
arbitrary number of Node objects, and a method to connect them.

The ``Hiveplot.connect()'' method draws edges as bezier curves. Start and
end points are set by the ``Axis.add_node()'' method, using the placement
information of the axis and a specified offset from its start point.

Control points are set at the same distance from the start point of an
axis as their corresponding nodes, but along an invisible axis that
shares its origin but diverges by a given angle.


\begin{figure}[!tpb]%figure1
  % \centerline{\includegraphics{fig01.eps}}
  \caption{Hiveplot of Erdos-Renyi graph trivially generated using
    NetworkX. Plot displays some SVG goodness: paths that make edges
    and axes can have any width, color or dash pattern, any valid SVG
    element can be used as a node. Define as many axes as you need,
    place them anywhere in the SVG canvas.}\label{fig:01}
\end{figure}

The ``Drawing'' structural component of the underlying svgwrite
library is available through the ``dwg'' property of Hiveplot, Axis
and Node objects. Any other shape or valid SVG element may be added to
them. This allows for a very flexible usage of marks, tags, ticks,
etc., as is shown in Figure \ref{fig:01}








%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%     please remove the " % " symbol from \centerline{\includegraphics{fig01.eps}}
%     as it may ignore the figures.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%






\section{Conclusion}

(Table~\ref{Tab:01}) Text Text Text Text Text Text  Text Text Text Text Text Text Text Text Text  Text Text Text Text Text Text. Figure \ref{fig:02} shows that the above method  Text Text Text Text  Text Text Text Text Text Text  Text Text.  \citealp{Boffelli03} might want to know about  text text text text


\section*{Acknowledgement}
Text Text Text Text Text Text  Text Text.  \citealp{Boffelli03} might want to know about  text text text text

\paragraph{Funding\textcolon} Text Text Text Text Text Text  Text Text.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%
%\bibliographystyle{plain}
%
%\bibliography{Document}


\begin{thebibliography}{}
\bibitem[Bofelli {\it et~al}., 2000]{Boffelli03} Bofelli,F., Name2, Name3 (2003) Article title, {\it Journal Name}, {\bf 199}, 133-154.

\bibitem[Bag {\it et~al}., 2001]{Bag01} Bag,M., Name2, Name3 (2001) Article title, {\it Journal Name}, {\bf 99}, 33-54.

\bibitem[Yoo \textit{et~al}., 2003]{Yoo03}
Yoo,M.S. \textit{et~al}. (2003) Oxidative stress regulated genes
in nigral dopaminergic neurnol cell: correlation with the known
pathology in Parkinson's disease. \textit{Brain Res. Mol. Brain
Res.}, \textbf{110}(Suppl. 1), 76--84.

\bibitem[Lehmann, 1986]{Leh86}
Lehmann,E.L. (1986) Chapter title. \textit{Book Title}. Vol.~1, 2nd edn. Springer-Verlag, New York.

\bibitem[Crenshaw and Jones, 2003]{Cre03}
Crenshaw, B.,III, and Jones, W.B.,Jr (2003) The future of clinical
cancer management: one tumor, one chip. \textit{Bioinformatics},
doi:10.1093/bioinformatics/btn000.

\bibitem[Auhtor \textit{et~al}. (2000)]{Aut00}
Auhtor,A.B. \textit{et~al}. (2000) Chapter title. In Smith, A.C.
(ed.), \textit{Book Title}, 2nd edn. Publisher, Location, Vol. 1, pp.
???--???.

\bibitem[Bardet, 1920]{Bar20}
Bardet, G. (1920) Sur un syndrome d'obesite infantile avec
polydactylie et retinite pigmentaire (contribution a l'etude des
formes cliniques de l'obesite hypophysaire). PhD Thesis, name of
institution, Paris, France.

\end{thebibliography}
\end{document}
